File: src/main/kotlin/com/github/itsempa/cyclamax/config/categories/Features.java
package com.github.itsempa.cyclamax.config.categories;

import com.github.itsempa.cyclamax.CyclaMax;
import com.google.gson.annotations.Expose;
import io.github.notenoughupdates.moulconfig.Config;
import io.github.notenoughupdates.moulconfig.annotations.Category;

public class Features extends Config {

    @Override
    public String getTitle() {
        return "CyclaMax " + CyclaMax.getVersion() + " by §cEmpa";
    }

    @Override
    public void saveNow() {
        CyclaMax.Companion.getManagedConfig().saveToFile();
    }

    @Expose
    @Category(name = "Cycla Box", desc = "")
    public CyclaBoxConfig cyclaBox = new CyclaBoxConfig();

    @Expose
    @Category(name = "Vampire Mask", desc = "")
    public VampireMaskConfig vampireMask = new VampireMaskConfig();

    @Expose
    @Category(name = "Kills Counter", desc = "")
    public KillsConfig killsCounter = new KillsConfig();

    @Expose
    @Category(name = "Misc", desc = "")
    public MiscConfig misc = new MiscConfig();
}

File: src/main/kotlin/com/github/itsempa/cyclamax/config/categories/Features.kt
package com.github.itsempa.cyclamax.config.categories

import at.hannibal2.skyhanni.deps.moulconfig.Config
import at.hannibal2.skyhanni.deps.moulconfig.annotations.Category
import com.github.itsempa.cyclamax.CyclaMax
import com.github.itsempa.cyclamax.CyclaMax.managedConfig
import com.google.gson.annotations.Expose

class Features : Config() {
    override fun getTitle(): String = "CyclaMax ${CyclaMax.VERSION} by §cEmpa"

    override fun saveNow() {
        managedConfig.saveToFile()
    }

    override fun shouldAutoFocusSearchbar(): Boolean = true

    @Expose
    @Category(name = "Cycla Box", desc = "")
    var cyclaBox: CyclaBoxConfig = CyclaBoxConfig()

    @Expose
    @Category(name = "Vampire Mask", desc = "")
    var vampireMask: VampireMaskConfig = VampireMaskConfig()

    @Expose
    @Category(name = "Kills Counter", desc = "")
    var killsCounter: KillsConfig = KillsConfig()

    @Expose
    @Category(name = "Misc", desc = "")
    var misc: MiscConfig = MiscConfig()
}
